sktime/registry/_tags.py

```diff
@@ -113,16 +113,62 @@
     ),
     (
         "y_inner_mtype",
-        "forecaster",
-        ("list", ["pd.Series", "pd.DataFrame", "np.array"]),
+        ["forecaster", "transformer"],
+        (
+            "list",
+            [
+                "pd.Series",
```

Comment:
since this is the types we can pass to `fit` - not necessarily the `_fit` internal types, is it fine to leave this with no change?

If we want to restrict the inner types, that would simply be a comment in the extension template, because it's a matter of the extender contract.